### ISSUE TITLE
Add support for passing library IDs with workflows and joining on that id

### ISSUE THREAD

**Issue by jashapiro** [AlexsLemonade/scpca-nf#8]
Joining runs with multimodal data by sample IDs works for most projects, but does not cover the case where a sample might have multiple libraries with separately paired RNA and CITEseq or cellhash data.  To support this, in https://github.com/AlexsLemonade/ScPCA-admin/pull/202, I added a new column, `scpca_library_id` which can be used as an unambiguous id for joining such libraries.

We could simply swap where we pass `sample_id` in the workflows, but a more sustainable method might be to use the system that `nf-core` seems to use: Rather than passing individual sample values (along with things like data files) as we do currently (which means we pass a bunch of values to each process), we could take advantage of a groovy `map` (dictionary in some other languages) to pass metadata values like various ids in a single process value (following `nf-core`, we can call this `meta`). This would (partially) free us from the tyranny of argument order, and allow easier changes to individual processes that might require more or different metadata, without having to change unaffected processes.

So basically, we would us `val(meta)` in place of `val(sample_id), val(run_id)` etc. and `meta` then use values such as `meta.sample_id` within processes where we would previously have just used `sample_id`. 

In the simplest implementation, we might use all info from the input csv file, but it is probably worth doing a bit of transformation to simplify some column names andonly retain columns that we need.

**Comment by jashapiro** [comment]
Note: this change could also be an opportunity to implement the suggestion raised here. https://github.com/AlexsLemonade/scpca-nf/pull/5#discussion_r681269764 to group outputs by run; we would instead be grouping by library, or we could group by sample/library (nested), which might be most useful for downstream processes.
